azure/services/managedclusters/managedclusters.go

```diff
@@ -173,6 +173,19 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 	}
 
+	if managedClusterSpec.Sku != nil {
+		var tier containerservice.ManagedClusterSKUTier
+		if managedClusterSpec.Sku.Tier == "Paid" {
+			tier = containerservice.ManagedClusterSKUTierPaid
+		} else {
+			tier = containerservice.ManagedClusterSKUTierFree
+		}
+		managedCluster.Sku = &containerservice.ManagedClusterSKU{
+			Name: containerservice.ManagedClusterSKUNameBasic,
+			Tier: tier,
+		}
```

Comment:
```suggestion
                skuName := containerservice.ManagedClusterSKUName(managedClusterSpec.Sku.Tier)
		managedCluster.Sku = &containerservice.ManagedClusterSKU{
			Name: skuName,
			Tier: tier,
		}
```

Reply:
@alexeldeib I believe that the `Name` field in `ManagedClusterSKU` is always [Basic](https://github.com/Azure/azure-sdk-for-go/blob/v55.8.0/services/containerservice/mgmt/2021-05-01/containerservice/enums.go#L210-L216), and we should set `Tier` according to the tier in our Spec. That's why I added the code here. Please let me know if that makes sense :)

Reply:
This is not correct, roundtrip conversion should not be lossy. See how it’s handled for the identity ref which is not present in v1a3. 

Stored the higher version and unmarshal it inside the conversion webhook, then rehydrate the field from the unmarshaled struct.

Reply:
Whoops, replied to the wrong comment.

Reply:
You’re correct about the code, I’m just simplifying the existing logic a bit. I think it’s the same?

Reply:
sorry but I'm not sure if I understand your suggested change, it looks like now the `Name` field is `skuName`, which is `containerservice.ManagedClusterSKUName(managedClusterSpec.Sku.Tier)`, but that isn't `containerservice.ManagedClusterSKUNameBasic`. Did I miss something here?

Reply:
my mistake -- I flipped tier and name in the suggestion. 

Basically, you are comparing tierName against a non-const string, but we already enforce the values coming in are enums exactly matching one of the supplied values. so we can just type coerce whatever is in the spec, rather than do a conditional check.

Reply:
updated the code snippet

Reply:
Got it, this make the code much simpler, thanks for the suggestion!